### DISCUSSION TITLE
Filter recordset before rendering in table

### DISCUSSION THREAD

**Discussion by photojo** [filamentphp/filament#5496] [Help]
I often use getTableFilters().

But now, I want to filter recordsets BEFORE shown in table, without changing my eloquent-query in getTableQuery()

Is this the right way? 
[https://filamentphp.com/docs/2.x/tables/testing#filters](https://filamentphp.com/docs/2.x/tables/testing#filters)

**Comment by danharrin** [comment]
Use a filter and pass a ->default() ?

**Reply by erdowaka** [reply]
can you explaint more sir ?

**Reply by dissto** [reply]
> can you explaint more sir ?

https://filamentphp.com/docs/3.x/tables/filters#applying-the-filter-by-default ?
